drivers/ksfilter/ks/device.c

```diff
@@ -518,23 +510,14 @@ IKsDevice_Pnp(
             }
 
             /* pass the irp down the driver stack */
-            Status = KspForwardIrpSynchronous(DeviceObject, Irp);
-
-            DPRINT1("IRP_MN_QUERY_INTERFACE Next Device: Status %x\n", Status);
-            Irp->IoStatus.Status = Status;
-            CompleteRequest(Irp, IO_NO_INCREMENT);
-            return Status;
+            IoSkipCurrentIrpStackLocation(Irp);
+            return IoCallDriver(DeviceHeader->KsDevice.NextDeviceObject, Irp);
         }
         case IRP_MN_QUERY_DEVICE_RELATIONS:
         {
             /* pass the irp down the driver stack */
-            Status = KspForwardIrpSynchronous(DeviceObject, Irp);
-
-            DPRINT("IRP_MN_QUERY_DEVICE_RELATIONS Next Device: Status %x\n", Status);
-
-            //Irp->IoStatus.Status = Status;
-            CompleteRequest(Irp, IO_NO_INCREMENT);
-            return Status;
+            IoSkipCurrentIrpStackLocation(Irp);
+            return IoCallDriver(DeviceHeader->KsDevice.NextDeviceObject, Irp);
```

Comment:
Nit: you could combine a bunch of these cases now, since they're identical. Or alternatively, leave the DPRINTs to keep them distinct.

Reply:
I think I'll leave it like that until a new person comes up to work on our audio stack :)